desc/grid.py

```diff
@@ -428,9 +428,17 @@ def _create_nodes(
             r = np.atleast_1d(rho)
             dr = np.zeros_like(r)
             if r.size > 1:
-                dr[0] = (r[0] + r[1]) / 2
-                dr[1:-1] = (r[2:] - r[:-2]) / 2
-                dr[-1] = 1 - (r[-2] + r[-1]) / 2
+                axis = np.isclose(r[0], 0)
+                # dr[0] = r[int(axis)]
+                # dr[1:-1] = (r[2:] - r[:-2]) / 2
+                # dr[-1] = 1 * r.size / (r.size - axis) - r[-1 - (not axis)]
```

Comment:
I actually think this is more correct. If the nodes aren't evenly spaced for some reason, this will correctly attribute half of the distance from `n-1` to `n` and from `n` to `n+1`

Reply:
Ok, I'll use switch it to this. Since the CI tests won't complete on the current run, I had the unit tests run locally and 218/218 passed.

Reply:
Making this change passes 217/218 unit tests, failing `test_vmec_save_1`. Looking into it